cmd/webhook/main.go

```diff
@@ -44,50 +59,60 @@ var (
 	}
 )
 
-var validationHook cmd.ValidatingAdmissionHook = handlers.NewFuncBackedValidator(logs.Log, GroupName, webhook.Scheme, validationFuncs)
-var mutationHook cmd.MutatingAdmissionHook = handlers.NewSchemeBackedDefaulter(logs.Log, GroupName, webhook.Scheme)
+var validationHook handlers.ValidatingAdmissionHook = handlers.NewFuncBackedValidator(logs.Log, GroupName, webhook.Scheme, validationFuncs)
+var mutationHook handlers.MutatingAdmissionHook = handlers.NewSchemeBackedDefaulter(logs.Log, GroupName, webhook.Scheme)
 
 func main() {
-	// Avoid "logging before flag.Parse" errors from glog
-	flag.CommandLine.Parse([]string{})
-
-	// parse the command line flags to pull out the tls-cert-file
-	// argument. This flag will be parsed by code inside cmd.RunAdmissionServer
-	// so no need to pass it through the call stack or have nice errors
-	tlsflagSet := flag.NewFlagSet("tls", flag.ContinueOnError)
-	tlsflagVal := tlsflagSet.String("tls-cert-file", "", "")
-	tlsflagSet.Parse(os.Args[1:])
-	if *tlsflagVal != "" {
-		runfilewatch(*tlsflagVal)
+	klog.InitFlags(flag.CommandLine)
+	flag.Parse()
+
+	log := klogr.New()
+	stopCh := setupSignalHandler()
+
+	var source server.CertificateSource
+	if tlsCertFile == "" || tlsKeyFile == "" {
+		log.Info("serving insecurely as tls certificate data not provided")
+	} else {
+		log.Info("enabling TLS as certificate file flags specified")
+		source = &server.FileCertificateSource{
+			CertPath: tlsCertFile,
+			KeyPath:  tlsKeyFile,
+			Log:      log,
+		}
 	}
 
-	cmd.RunAdmissionServer(
-		validationHook,
-		mutationHook,
-	)
+	srv := server.Server{
+		ListenAddr:        fmt.Sprintf(":%d", securePort),
+		HealthzAddr:       fmt.Sprintf(":%d", healthzPort),
+		CertificateSource: source,
+		ValidationWebhook: validationHook,
+		MutationWebhook:   mutationHook,
+		Log:               log,
+	}
+	if err := srv.Run(stopCh); err != nil {
+		log.Error(err, "error running server")
+		os.Exit(1)
+	}
 }
 
-func runfilewatch(filename string) {
-	info, err := os.Stat(filename)
-	if err != nil {
-		// missing TLS cert file will get turned into a proper error later
-		return
-	}
-	modtime := info.ModTime()
+var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+var onlyOneSignalHandler = make(chan struct{})
+
+// setupSignalHandler registered for SIGTERM and SIGINT. A stop channel is returned
+// which is closed on one of these signals. If a second signal is caught, the program
+// is terminated with exit code 1.
+func setupSignalHandler() (stopCh <-chan struct{}) {
+	close(onlyOneSignalHandler) // panics when called twice
```

Comment:
Is there not a nicer way of doing? Guess it doesn't matter too much...

Reply:
This was taken from the our current controller's `cmd` package 🤷‍♂ (and taken from k8s before that)